builder/qemu/step_run_test.go

```diff
@@ -105,7 +105,7 @@ func Test_UserOverrides(t *testing.T) {
 			"-fda", "fake_floppy_path",
 			"-name", "myvm",
 			"-netdev", "user,id=user.0,hostfwd=tcp::5000-:0",
-			"-vnc", ":5905",
+			"-vnc", ":5",
```

Comment:
What happens if the VNC port is greater than 5999?  Is this also invalid?

Reply:
I tested with setting vnc_min_port to 30000 and vnc_max_port to 40000 and it works okay AFAIK. Any valid port above 5900 should be fine, QEMU's code says it should just error out if you try to pass it an invalid port. 

```
qemu-builder: Found port for communicator (SSH, WinRM, etc): 3432.
qemu-builder: Looking for available port between 30000 and 40000 on 0.0.0.0
qemu-builder: Starting VM, booting disk image
qemu-builder: The VM will be run headless, without a GUI. If you want to
qemu-builder: view the screen of the VM, connect via VNC without a password to
qemu-builder: vnc://0.0.0.0:34807
qemu-builder: Waiting 10s for boot...
qemu-builder: Connecting to VM via VNC (0.0.0.0:34807)
qemu-builder: Typing the boot command over VNC...
qemu-builder: Not using a NetBridge -- skipping StepWaitGuestAddress
qemu-builder: Using ssh communicator to connect: 127.0.0.1
qemu-builder: Waiting for SSH to become available...
```

That said, packer will sit there stuck if you try to use an invalid port anyways until you kill it. And setting the port that high is a very silly thing to do. :) 

```
qemu-builder: Found port for communicator (SSH, WinRM, etc): 3948.
==> qemu-builder: Looking for available port between 300000 and 300002 on 0.0.0.0
Cancelling build after receiving interrupt
==> qemu-builder: Error finding VNC port: port 300001 cannot be opened: listen tcp: address 300001: invalid port
==> qemu-builder: Deleting output directory...
Build 'qemu-builder' errored after 1 minute 8 seconds: Error finding VNC port: port 300001 cannot be opened: listen tcp: address 300001: invalid port
```

FWIW If you're interested, you can actually see QEMU's handling for the port here under 'vnc_display_get_address' function. 

https://github.com/qemu/qemu/blob/1f7c02797fa189ce4b34382020bbce63262a5758/ui/vnc.c#L3581

Reply:
> And setting the port that high is a very silly thing to do. :)

With a user base as big as ours, I tend to try to avoid making assumptions about what people will consider silly :D

Reply:
Hahahahaha yeah with you on that one!  I'll toss in a sanity check in config.go for max ports then when I get the chance then.

Reply:
With 8ef4cfa0

```
vnc_port_min = 65536
vnc_port_max = 65535 

Error: Failed to prepare build: "qemu-builder"

2 error(s) occurred:

* vmc_port_min and vnc_port_max must both be below 65535 to be valid TCP ports
* vnc_port_min must be less than vnc_port_max
```

```
vnc_port_min = 65535
vnc_port_max = 65535
 
qemu-builder: Found port for communicator (SSH, WinRM, etc): 3860.
qemu-builder: Looking for available port between 65535 and 65535 on 0.0.0.0
qemu-builder: Starting VM, booting disk image
qemu-builder: The VM will be run headless, without a GUI. If you want to
qemu-builder: view the screen of the VM, connect via VNC without a password to
qemu-builder: vnc://0.0.0.0:65535
qemu-builder: Waiting 10s for boot...
qemu-builder: Connecting to VM via VNC (0.0.0.0:65535)
qemu-builder: Typing the boot command over VNC...
```